src/main/java/fr/insee/rmes/services/datasets/DataSetsImpl.java

```diff
@@ -88,13 +90,26 @@ private JsonNode emptyDataSetModelSwagger(DataSetModelSwagger dataSetModelSwagge
         return dataSetFinalNode;
     }
 
+    public JSONObject formattedJsonDataset(JSONObject givenJson) {
+        JSONObject jsonResponse = new JSONObject();
+        for (PatchDatasetProperties property : PatchDatasetProperties.values()) {
+            String key = property.toString();
+            if (givenJson.has(key)) {
+                Object value = givenJson.get(key);
+                jsonResponse.put(key, value);
+            }
+        }
+        return jsonResponse;
+    }
+
     @Override
-    public String patchDataset(String datasetId, String observationNumber, String token) throws RmesException {
+    public String patchDataset(String datasetId, String stringPatchDataset, String token) throws RmesException, MalformedURLException {
         try {
```

Comment:
Gérer le cas où aucun des champs possibles n'est renseigné : retourner un code HTTP 400
Actuellement, renvoie un 200 avec dans le message (entre autres) :
  "status": 404,
  "error": "Not Found",